applications/FluidDynamicsApplication/custom_elements/data_containers/qs_vms/qs_vms_adjoint_element_data.h

```diff
@@ -0,0 +1,151 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_QS_VMS_ADJOINT_ELEMENT_DATA_H)
+#define KRATOS_QS_VMS_ADJOINT_ELEMENT_DATA_H
+
+// System includes
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "custom_elements/data_containers/derivatives.h"
+#include "custom_elements/data_containers/qs_vms/qs_vms_derivative_utilities.h"
+#include "custom_elements/data_containers/qs_vms/qs_vms_residual_derivatives.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+template <unsigned int TDim, unsigned int TNumNodes>
+class QSVMSAdjointElementData
+{
+private:
+    ///@name Private Type Definitions
+    ///@{
+
+    using IndexType = std::size_t;
+
+    using TResidualsDerivatives = QSVMSResidualDerivatives<TDim, TNumNodes>;
+
+    using Data = typename TResidualsDerivatives::Data;
+
+    using ResidualsContributions = typename TResidualsDerivatives::ResidualsContributions;
+
+    using PressureDerivativeContributions = typename TResidualsDerivatives::template VariableDerivatives<typename QSVMSDerivativeUtilities<TDim>::template PressureDerivative<TNumNodes>>;
+
+    template<unsigned int TDirectionIndex>
+    using VelocityDerivativeContributions = typename TResidualsDerivatives::template VariableDerivatives<typename QSVMSDerivativeUtilities<TDim>::template VelocityDerivative<TNumNodes, TDirectionIndex>>;
+
+    template<unsigned int TDirectionIndex>
+    using ShapeDerivatives = typename TResidualsDerivatives::template VariableDerivatives<typename QSVMSDerivativeUtilities<TDim>::template ShapeDerivative<TNumNodes, TDirectionIndex>>;
+
+    template<unsigned int TDirectionIndex>
+    using AccelerationDerivativeContributions = typename TResidualsDerivatives::template SecondDerivatives<TDirectionIndex>;
+
+    static constexpr IndexType ElementDataContainerIndex = 0;
+
+    static constexpr IndexType ResidualColumnOffset = 0;
+
+    ///@}
+
+public:
+    ///@name Template Type Definitions
+    ///@{
+
```

Comment:
Would you mind formatting this a bit? It's hard to follow...

Reply:
I will :)

Reply:
done 👍